llm4pddl/utils.py

```diff
@@ -238,6 +241,75 @@ def minify_pddl_problem(problem: str) -> str:
     return new_problem
 
 
+def embed_training_tasks(training_tasks: Sequence[Task]) -> List:
+    """"embeds a list of tasks.
+
+    Returns a list of embeddings with indicies corresponding to its
+    task.
+    """
+    embedding_model = SentenceTransformer(FLAGS.embedding_model_name)
+    embeddings = [embed_task(task, embedding_model) for task in training_tasks]
+    return embeddings
+
+
+def embed_task(task: Task, embedding_model: SentenceTransformer) -> List:
+    """Embeds a task using embedding_model.
+
+    Returns a numpy array?
+    """
+    with open(task.problem_file, 'r', encoding='utf-8') as f:
+        task_string = f.read()
+    task_string = minify_pddl_problem(task_string)
+    embedding = embedding_model.encode(task_string)
+    return embedding
+
+
+def make_embeddings_mapping(embeddings: Sequence,
+                            dataset: Dataset) -> List[Dict]:
+    """Makes embeddings mapping for training data."""
+    assert len(embeddings) == len(dataset)
+    return [{
+        'embedding': emb,
+        'datum': datum
+    } for emb, datum in zip(embeddings, dataset)]
+
+
+def get_closest(task: Task, embeddings_mapping: List[Dict],
+                num_train: int) -> List:
```

Comment:
let's call this `num_embeddings` or `num_closest` or `top_n` instead of `num_train`